### PR TITLE
Admin page to get a random active user

### PR DESCRIPTION
**Pull request by jpaddison3**
Then you can message them and bada bing bada boom you have a user interview. (JP building a feature for himself.)

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204858390195940) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/admin/RandomUserPage.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      const user: UsersMinimumInfo|undefined = data?.GetRandomUser;
+      if (!user) {
+        // eslint-disable-next-line no-console
+        console.error("No user found");
+        return;
+      }
+      // Redirect to the user page
+      const url = `/users/${user.slug}`;
+      if (newTabKeyHeld) {
+        window.open(url, "_blank")?.focus();
```
Comment (jpaddison3) [inline packages/lesswrong/components/admin/RandomUserPage.tsx:61]: This whole useEffect crap is for this line, which I couldn't figure out how to do any more elegant way.

The .focus() is intended to keep the focus on the original tab, but it doesn't work, and apparently Chrome doesn't want to allow JS to open tabs in the background so it's not possible? I decided I didn't care.

Comment (oetherington) [reply packages/lesswrong/components/admin/RandomUserPage.tsx:61]: This seems convoluted but it's not immediately obvious to me how to get this behaviour any other way, so I guess it's fine 🤷‍♂️

File: packages/lesswrong/server/repos/UsersRepo.ts [hunk 2]
```diff
@@ -191,4 +191,44 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
     const {count} = await this.getRawDb().one(`SELECT COUNT(*) FROM "Users"`);
     return count;
   }
+  
+  async getRandomActiveUser(): Promise<DbUser> {
+    return this.one(`
+      SELECT u.*
```
Comment (oetherington) [inline packages/lesswrong/server/repos/UsersRepo.ts:197]: Part of me feels like we should have a warning on these functions that they're going to be _real_ slow. Like, it's fine for this page, but I don't want someone to come along later and use them in a performance critical code path. That might just be over the top though.

**Review by oetherington — APPROVED** [review]
🦭 LGTM

File: packages/lesswrong/server/repos/UsersRepo.ts (hunk 2 above)
Comment (jpaddison3) [reply packages/lesswrong/server/repos/UsersRepo.ts:197]: Seemed acceptable (this is prod):

```
 Limit  (cost=61417.39..61417.39 rows=1 width=4197) (actual time=428.624..428.687 rows=1 loops=1)
   ->  Sort  (cost=61417.39..61419.89 rows=1001 width=4197) (actual time=428.623..428.685 rows=1 loops=1)
         Sort Key: (random())
         Sort Method: top-N heapsort  Memory: 29kB
         ->  Nested Loop  (cost=50486.49..61412.39 rows=1001 width=4197) (actual time=364.414..417.635 rows=4689 loops=1)
               ->  Finalize GroupAggregate  (cost=50486.08..52053.00 rows=2002 width=26) (actual time=364.338..392.588 rows=4745 loops=1)
                     Group Key: "ReadStatuses"."userId"
                     Filter: (max("ReadStatuses"."lastUpdated") > (now() - '1 mon'::interval))
                     Rows Removed by Filter: 22599
                     ->  Gather Merge  (cost=50486.08..51887.81 rows=12014 width=26) (actual time=364.309..381.907 rows=52278 loops=1)
                           Workers Planned: 2
                           Workers Launched: 2
                           ->  Sort  (cost=49486.06..49501.07 rows=6007 width=26) (actual time=355.954..358.205 rows=17426 loops=3)
                                 Sort Key: "ReadStatuses"."userId"
                                 Sort Method: quicksort  Memory: 2325kB
                                 Worker 0:  Sort Method: quicksort  Memory: 2052kB
                                 Worker 1:  Sort Method: quicksort  Memory: 1629kB
                                 ->  Partial HashAggregate  (cost=49048.97..49109.04 rows=6007 width=26) (actual time=318.012..321.160 rows=17426 loops=3)
                                       Group Key: "ReadStatuses"."userId"
                                       Batches: 1  Memory Usage: 3089kB
                                       Worker 0:  Batches: 1  Memory Usage: 2833kB
                                       Worker 1:  Batches: 1  Memory Usage: 2577kB
                                       ->  Parallel Seq Scan on "ReadStatuses"  (cost=0.00..44019.98 rows=1005798 width=26) (actual time=0.005..82.533 rows=802164 loops=3)
               ->  Index Scan using "Users_pkey" on "Users" u  (cost=0.41..4.66 rows=1 width=4189) (actual time=0.004..0.004 rows=1 loops=4745)
                     Index Cond: ((_id)::text = ("ReadStatuses"."userId")::text)
                     Filter: (NOT COALESCE(deleted, false))
                     Rows Removed by Filter: 0
 Planning Time: 2.938 ms
 Execution Time: 429.211 ms
(29 rows)
```

**Pull request merged**